### PR TITLE
Serpent5 scoped di/ra

### LINKED ISSUE #20401: Better documentation for scope DI usage with HttpClientFactory scenarios
> The IHttpClientFactory creates a separate DI scope for each handler. Handlers can depend upon services of any scope. Services that handlers depend upon are disposed when the handler is disposed.

This statement should be more visible (like a **NOTE** in yellow in bold). We should also show examples of how scoped services resolved. Show the same scoped service being resolve in an HttpMessageHandler and the request with difference lifetimes. That's bitten some people.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 881cef77-11ef-ac11-06d0-0996a1ccfa1b
* Version Independent ID: 3d356334-e672-7e8b-1fef-7944ebee99eb
* Content: [Make HTTP requests using IHttpClientFactory in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/http-requests?view=aspnetcore-3.1#outgoing-request-middleware)
* Content Source: [aspnetcore/fundamentals/http-requests.md](https://github.com/dotnet/AspNetCore.Docs/blob/master/aspnetcore/fundamentals/http-requests.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-fundamentals**
* GitHub Login: @stevejgordon
* Microsoft Alias: **scaddie**

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Fixes #20401. Moved from #20458

@serpent5  put together an example and added a TODO into the topic for explanation. I'll do that in the next commit, but I'm offering this up here as a draft for some feedback on the approach.

[Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/fundamentals/http-requests?view=aspnetcore-6.0&branch=pr-en-us-21275#use-di-in-outgoing-request-middleware)

Public review of doc build 
- download https://www.dropbox.com/sh/yy994551cje0kap/AAB0tRnm7gze2kYrSYPSFHIda?dl=0
- unzip
- view HTML file in FF (Fire Fox). It best to view the doc with the widest window that has no left or right TOC (Table  of contents)

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
@mkArtakMSFT please assign someone to review this fantastic addition by @serpent5 

@serpent5 review my changes:
* Appended  `.SetHandlerLifetime(TimeSpan.FromSeconds(5));` because I'm too impatient to wait the default 120 seconds.
* Copy/pasted your comments on how to test the app.
* Show entire `ConfigureServices` rather than just `services.AddScoped<IOperationScoped, OperationScoped>();`

File: aspnetcore/fundamentals/http-requests.md [hunk 1]
```diff
@@ -261,13 +260,13 @@ For example, consider the following interface and its implementation, which repr
...
 
-[!code-csharp[](http-requests/samples/3.x/HttpRequestsSample/Startup.cs?name=snippet_IOperationScoped)]
+[!code-csharp[](http-requests/samples/3.x/HttpRequestsSample/Startup.cs?name=snippet_IOperationScoped&highlight=18,26)]
 
 The following delegating handler consumes and uses `IOperationScoped` to set the `X-OPERATION-ID` header for the outgoing request:
 
 [!code-csharp[](http-requests/samples/3.x/HttpRequestsSample/Handlers/OperationHandler.cs?name=snippet_Class&highlight=13)]
 
-<!-- TODO: Bring it all home -->
+In the [`HttpRequestsSample` download](https://github.com/dotnet/AspNetCore.Docs/tree/master/aspnetcore/fundamentals/http-requests/samples/3.x/HttpRequestsSample)], navigate to `/Operation` and refresh the page. The request scope value changes for each request, but the handler scope value only change every 5 seconds.
```
Comment (serpent5) [inline aspnetcore/fundamentals/http-requests.md:269]:
```suggestion
In the [`HttpRequestsSample` download](https://github.com/dotnet/AspNetCore.Docs/tree/master/aspnetcore/fundamentals/http-requests/samples/3.x/HttpRequestsSample)], navigate to `/Operation` and refresh the page. The request scope value changes for each request, but the handler scope value only changes every 5 seconds.
```

File: aspnetcore/fundamentals/http-requests/samples/3.x/HttpRequestsSample/Startup.cs [hunk 2]
```diff
@@ -25,24 +26,25 @@ public void ConfigureServices(IServiceCollection services)
                 // For sample purposes, assume TodoClient is used in the context of an incoming request.
```
Comment (serpent5) [inline aspnetcore/fundamentals/http-requests/samples/3.x/HttpRequestsSample/Startup.cs:26]:
```suggestion
```

File: aspnetcore/fundamentals/http-requests/samples/3.x/HttpRequestsSample/Startup.cs [hunk 3]
```diff
@@ -25,24 +26,25 @@ public void ConfigureServices(IServiceCollection services)
...
                 var httpRequest = sp.GetRequiredService<IHttpContextAccessor>().HttpContext.Request;
 
-                httpClient.BaseAddress = new Uri(UriHelper.BuildAbsolute(httpRequest.Scheme, httpRequest.Host, httpRequest.PathBase));
+                httpClient.BaseAddress = new Uri(UriHelper.BuildAbsolute(httpRequest.Scheme,
```
Comment (serpent5) [inline aspnetcore/fundamentals/http-requests/samples/3.x/HttpRequestsSample/Startup.cs:29]:
```suggestion
                // For sample purposes, assume TodoClient is used in the context of an incoming request.
                httpClient.BaseAddress = new Uri(UriHelper.BuildAbsolute(httpRequest.Scheme,
```

With this code being shown in a snippet, I think the comment should live closer to this line.

**Review by serpent5 — COMMENTED** [review]
Awesome. Thanks for taking over on this one.

The `SetHandlerLifetime` change was a good call.

**Comment by Rick-Anderson** [comment]
@serpent5  I applied your suggestions and updated the public preview.

**Comment by Rick-Anderson** [comment]
Awesome work @serpent5

**Pull request merged**